### PR TITLE
Cherry pick Order case sensitivity changes to 3.4

### PR DESCRIPTION
**Pull request by karuppayya**
CP-ed commit https://github.com/apache/iceberg/commit/6504fd809199cbc88ef890cc2f5b42a2cf00e102

### CODE REVIEW

**Comment by karuppayya** [comment]
@RussellSpitzer @aokolnychyi

**Comment by RussellSpitzer** [comment]
Weird Project Nessie Dependency failure? Maybe just need to rebase to master?

File: spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java [hunk 1]
```diff
@@ -64,6 +67,28 @@ public void testSetWriteOrderByColumn() {
...
+              sql("ALTER TABLE %s WRITE ORDERED BY category, id", tableName);
+            })
+        .isInstanceOf(ValidationException.class);
+
+    sql("SET %s=false", SQLConf.CASE_SENSITIVE().key());
+    Assertions.assertThatNoException()
+        .isThrownBy(
+            () -> {
+              sql("ALTER TABLE %s WRITE ORDERED BY category, id", tableName);
+            });
```
Comment (RussellSpitzer) [inline spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java:89]: I would prefer we just run the Alter and then check that the sortOrder is correctly set. Personally, I'm generally not a fan of "did this not throw an exception" tests.

File: spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java [hunk 2]
```diff
@@ -64,6 +67,28 @@ public void testSetWriteOrderByColumn() {
...
 
+  @Test
+  public void testSetWriteOrderWithCaseSensitiveColumnNames() {
+    sql(
+        "CREATE TABLE %s (Id bigint NOT NULL, Category string, ts timestamp, data string) USING iceberg",
+        tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    Assert.assertTrue("Table should start unsorted", table.sortOrder().isUnsorted());
+    sql("SET %s=true", SQLConf.CASE_SENSITIVE().key());
+    Assertions.assertThatThrownBy(
```
Comment (RussellSpitzer) [inline spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java:78]: Can we add a message check to this, just want to make sure the validation exception we get is related to the names not matching

**Review by RussellSpitzer — APPROVED** [review]

**Review by RussellSpitzer — APPROVED** [review]

**Pull request merged**